ivy/__init__.py

```diff
@@ -202,17 +202,37 @@ def __new__(cls, shape_tup):
                 current_backend(shape_tup).NativeArray,
             )
         ivy.utils.assertions.check_isinstance(shape_tup, valid_types)
-        if isinstance(shape_tup, int):
-            shape_tup = (shape_tup,)
+        if isinstance(shape_tup, ivy.NativeShape):
+            self._shape = shape_tup
+        elif isinstance(shape_tup, int):
+            self._shape = (shape_tup,)
         elif isinstance(shape_tup, list):
-            shape_tup = tuple(shape_tup)
-        ivy.utils.assertions.check_all(
-            [isinstance(v, int) or ivy.is_int_dtype(v.dtype) for v in shape_tup],
-            "shape must take integers only",
+            self._shape = tuple(shape_tup)
+        else:
+            self._shape = None
+
+    def __repr__(self):
+        return (
+            f"Ivy.Shape{self._shape}" if self._shape is not None else "Ivy.Shape(None)"
```

Comment:
I guess that this would be a bit weird with tf and torch? not sure but the output would be something like `ivy.Shapetorch.Shape(1, 2, 3)`?